components/Forms/SignUp.js

```diff
@@ -0,0 +1,187 @@
+import { useReducer } from 'react';
+import Link from 'next/link';
+
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import { Link as MUILink } from '@mui/material';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+
+const formReducer = (state, action) => {
+	switch (action.type) {
+		case 'FORM_CHANGE':
+			const newState = {
+				controls: {
+					...state.controls,
+					[action.control]: {
+						value: action.value,
+						isValid: action.isValid,
+						isTouched: true,
+					},
+				},
+			};
+			newState.isValid = Object.values(newState.controls).every(
+				control => control.isValid,
+			);
+			return newState;
+		default:
+			return state;
+	}
+};
+
+const SignUp = () => {
+	const [form, dispatch] = useReducer(formReducer, {
+		controls: {
+			email: { value: '', isValid: false, isTouched: false },
+			password: { value: '', isValid: false, isTouched: false },
+			firstName: { value: '', isValid: false, isTouched: false },
+			lastName: { value: '', isValid: false, isTouched: false },
+		},
+		isValid: false,
+	});
+
+	const handleSubmit = event => {
+		event.preventDefault();
+		console.log('submit');
+	};
+
+	const onChangeHandler = (event, controlName) => {
+		const value = event.target.value;
+		let isValid = false;
+		switch (controlName) {
+			case 'email':
+				isValid = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);
+				break;
+			case 'password':
+				isValid = value.trim().length >= 6;
+				break;
+			case 'firstName':
+				isValid = value.trim().length > 0;
+				break;
+			case 'lastName':
+				isValid = value.trim().length > 0;
+				break;
+		}
+
+		dispatch({ type: 'FORM_CHANGE', control: controlName, value, isValid });
+	};
+
+	return (
+		<Container component="main" maxWidth="xs">
+			<CssBaseline />
```

Comment:
No need to add CssBaseline on each page. It is already included globally in `_document.js`

Reply:
Oops sorry I didn't noticed CssBaseline in ```_document.js```